### PR TITLE
Maid 1024 pure sentinel

### PR DESCRIPTION
**Pull request by inetic**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/sentinel/17)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by maidsafe-highfive** [comment]
OS X Build failed

**Comment by maidsafe-highfive** [comment]
Windows Build failed

**Comment by maidsafe-highfive** [comment]
OS X Build Successful

File: Cargo.toml [hunk 1]
```diff
@@ -16,5 +16,8 @@ rustc-serialize = "*"
 cbor = "*"
 sodiumoxide = "0.0.5"
 rand = "*"
-accumulator = "0.0.1"
+#accumulator = "0.0.1"
 lru_time_cache = "0.1.6"
+
+[dependencies.accumulator]
+git = "https://github.com/inetic/accumulator"
```
Comment (benjaminbollen) [inline Cargo.toml:23]: Can you link a PR to accumulator?

File: src/sentinel.rs [hunk 2]
```diff
@@ -82,44 +73,41 @@ impl<'a, Request, Name>
...
-            None => {
-                self.sender.get_signing_keys(request.get_source());
-                self.claim_accumulator.add(request, (claimant, signature, claim));
-                return None;
+    /// This adds a new claim for the provided request. The claimant name and
+    /// the signature provided will be used to verify the claim with the keys
+    /// that are independently retrieved. When an added claim leads to the
+    /// resolution of the request, the request and the claim are returned.
+    /// All resolved claims have to be identical. Otherwise None is returned.
+    pub fn add_claim(&mut self,
```
Comment (benjaminbollen) [inline src/sentinel.rs:96]: Maybe mention in the documentation also that RequestKeys(Name) is returned when Sentinel requires the keys

File: src/sentinel.rs [hunk 3]
```diff
@@ -128,104 +116,63 @@ impl<'a, Request, Name>
...
-            .collect::<Vec<&SerialisedClaim>>();
-        assert_eq!(resolved_claims.len(), 1);
-
-        let resolved_claim : Option<SerialisedClaim>
-            = match resolved_claims.first() {
-            Some(&serialised_claim) => Some(serialised_claim.clone()),
-            None => None
-        };
-        resolved_claim
+            .nth(0)
```
Comment (benjaminbollen) [inline src/sentinel.rs]: can we maybe keep the assert .len() == 1 ? with a .inspect() perhaps?

File: Cargo.toml (hunk 1 above)
Comment (inetic) [inline Cargo.toml:23]: https://github.com/maidsafe/accumulator/pull/7

File: src/sentinel.rs (hunk 2 above)
Comment (inetic) [inline src/sentinel.rs:96]: Will do

File: src/sentinel.rs (hunk 3 above)
Comment (inetic) [inline src/sentinel.rs]: ok

**Comment by benjaminbollen** [comment]
great work.  will merge when those minor remarks are pushed

**Comment by inetic** [comment]
I think you're right. I must have gotten confused (I blame it on me working late :) ). I guess I should use the `debug_assert` macro instead of the `assert` one then (apart from switching back to using the Frequency structure).

**Comment by benjaminbollen** [comment]
perfect, thanks Peter!  great work

**Pull request merged**